### PR TITLE
Adds workaround for Async Multipart uploads greater than 25 kb

### LINKED ISSUE #560: Async request goes not work when uploading files greater than 25 KB
When sending multipart put request with file greater than 25 KB I'm getting `Content length is too long: 1300881` exception coming from `org.apache.http.entity.mime.MultipartFormEntity.getContent(MultipartFormEntity.java:103)`

Stack trace:
````
failed:645, core$request$reify__18941 (clj_http)
failed:138, BasicFuture (org.apache.http.concurrent)
executionFailed:101, DefaultClientExchangeHandlerImpl (org.apache.http.impl.nio.client)
failed:426, AbstractClientExchangeHandler (org.apache.http.impl.nio.client)
exception:155, HttpAsyncRequestExecutor (org.apache.http.nio.protocol)
produceOutput:310, DefaultNHttpClientConnection (org.apache.http.impl.nio)
onOutputReady:86, InternalIODispatch (org.apache.http.impl.nio.client)
onOutputReady:39, InternalIODispatch (org.apache.http.impl.nio.client)
outputReady:152, AbstractIODispatch (org.apache.http.impl.nio.reactor)
writable:188, BaseIOReactor (org.apache.http.impl.nio.reactor)
processEvent:341, AbstractIOReactor (org.apache.http.impl.nio.reactor)
processEvents:315, AbstractIOReactor (org.apache.http.impl.nio.reactor)
execute:276, AbstractIOReactor (org.apache.http.impl.nio.reactor)
execute:104, BaseIOReactor (org.apache.http.impl.nio.reactor)
run:588, AbstractMultiworkerIOReactor$Worker (org.apache.http.impl.nio.reactor)
run:834, Thread (java.lang)
````

````
getContent():103, MultipartFormEntity (org.apache.http.entity.mime), MultipartFormEntity.java
produceContent(ContentEncoder, IOControl):65, EntityAsyncContentProducer (org.apache.http.nio.entity), EntityAsyncContentProducer.java
produceContent(ContentEncoder, IOControl):125, BasicAsyncRequestProducer (org.apache.http.nio.protocol), BasicAsyncRequestProducer.java
produceContent(InternalState, ContentEncoder, IOControl):262, MainClientExec (org.apache.http.impl.nio.client), MainClientExec.java
produceContent(ContentEncoder, IOControl):140, DefaultClientExchangeHandlerImpl (org.apache.http.impl.nio.client), DefaultClientExchangeHandlerImpl.java
outputReady(NHttpClientConnection, ContentEncoder):241, HttpAsyncRequestExecutor (org.apache.http.nio.protocol), HttpAsyncRequestExecutor.java
produceOutput(NHttpClientEventHandler):290, DefaultNHttpClientConnection (org.apache.http.impl.nio), DefaultNHttpClientConnection.java
onOutputReady(DefaultNHttpClientConnection):86, InternalIODispatch (org.apache.http.impl.nio.client), InternalIODispatch.java
onOutputReady(Object):39, InternalIODispatch (org.apache.http.impl.nio.client), InternalIODispatch.java
outputReady(IOSession):152, AbstractIODispatch (org.apache.http.impl.nio.reactor), AbstractIODispatch.java
writable(SelectionKey):188, BaseIOReactor (org.apache.http.impl.nio.reactor), BaseIOReactor.java
processEvent(SelectionKey):341, AbstractIOReactor (org.apache.http.impl.nio.reactor), AbstractIOReactor.java
processEvents(Set):315, AbstractIOReactor (org.apache.http.impl.nio.reactor), AbstractIOReactor.java
execute():276, AbstractIOReactor (org.apache.http.impl.nio.reactor), AbstractIOReactor.java
execute(IOEventDispatch):104, BaseIOReactor (org.apache.http.impl.nio.reactor), BaseIOReactor.java
run():588, AbstractMultiworkerIOReactor$Worker (org.apache.http.impl.nio.reactor), AbstractMultiworkerIOReactor.java
run():834, Thread (java.lang), Thread.java
````

The 25 kb limitation in `MultipartFormEntity#getContent` is discussed [here](https://stackoverflow.com/questions/51435314/why-multipartentitybuilder-throws-org-apache-http-contenttoolongexception).

### PR DESCRIPTION
**Pull request by rymndhng**
Fixes #560

### CODE REVIEW

File: src/clj_http/multipart.clj [hunk 1]
```diff
@@ -125,6 +125,28 @@
...
+    (isStreaming [_] (.isStreaming mp-entity))
+    (getContentLength [_] (.getContentLength mp-entity))
+    (getContentType [_] (.getContentType mp-entity))
+    (getContentEncoding [_] (.getContentEncoding mp-entity))
+    (consumeContent [_] (.consumeContent mp-entity))
+    (getContent [_]
+      (let [os (java.io.ByteArrayOutputStream.)]
+        (.writeTo mp-entity os)
+        (.flush os)
+        (java.io.ByteArrayInputStream. (.toByteArray os))))
```
Comment (rymndhng) [inline src/clj_http/multipart.clj:147]: I've removed these lines of code: https://github.com/apache/httpcomponents-client/blob/4.5.x/httpmime/src/main/java/org/apache/http/entity/mime/MultipartFormEntity.java#L100-L104

**Review by dakrone — APPROVED** [review]
LGTM

**Comment by dakrone** [comment]
It does appear there is a suggested workaround in the JIRA issue by wrapping with a `BufferedHttpEntity`, perhaps that would be a better solution for this?

**Comment by rymndhng** [comment]
@dakrone I gave it a try, but unfortunately it fails for the same reason in my test case. The `BufferedHttpEntity` may buffer the contents, depending on some of the flags. 

See https://github.com/apache/httpcomponents-core/blob/cd88841e0/httpcore/src/main/java/org/apache/http/entity/BufferedHttpEntity.java#L58-L68

Removing the checks in the original solution is more reliable to retain the existing behaviour our users expect.

**Comment by romashatomar** [comment]
Hi,
I am facing "Content too long" exception when trying to update content as a byte array (using multi part form data) using Http Async Client . I see that changes have been made to remove those lines of code . Have they been released ? I am using  httpmime : 4.5.12  and I get the same error.

**Comment by rymndhng** [comment]
@romashatomar the changes were released in clj-http `3.11.0`. If you are seeing issues with clj-http, file a new issue and provide reproduction steps and I can help investigate further.

**Pull request merged**